lighthouse-core/test/lib/tracehouse/trace-processor-test.js

```diff
@@ -322,6 +324,31 @@ describe('TraceProcessor', () => {
       });
     });
 
+    describe('finds correct LCP', () => {
+      it('if there was a tracingStartedInPage after the frame\'s navStart', () => {
+        const trace = TraceProcessor.computeTraceOfTab(lcpTrace);
+        assert.equal(trace.mainFrameIds.frameId, '906A10385298DD996B521026AF4DA204');
+        assert.equal(trace.navigationStartEvt.ts, 1671221915754);
+        assert.equal(trace.firstContentfulPaintEvt.ts, 1671226617803);
+        assert.equal(trace.largestContentfulPaintEvt.ts, 1671236939268);
+        assert.ok(!trace.invalidLcp);
+      });
+
+      it('invalidates if last event is ::Invalidate', () => {
+        const invalidLcpTrace = createTestTrace({navigationStart: 0, traceEnd: 2000});
+        const frame = invalidLcpTrace.traceEvents[0].args.frame;
+        const args = {frame};
+        const cat = 'loading,rail,devtools.timeline';
+        invalidLcpTrace.traceEvents.push(...[
+          {name: 'largestContentfulPaint::Candidate', cat, args, ts: 1000, duration: 10},
+          {name: 'largestContentfulPaint::Invalidate', cat, args, ts: 1100, duration: 10},
+        ]);
+        const trace = TraceProcessor.computeTraceOfTab(invalidLcpTrace);
+        assert.equal(trace.largestContentfulPaintEvt, undefined);
+        assert.ok(trace.invalidLcp);
+      });
+    });
```

Comment:
need tests for
- no LCP event at all
- only LCP before navstart
- and maybe a Candidate, Invalidate, Candidate case (makes sure we're actually looking for the last one (based on the ts), that the invalidate doesn't screw it up, etc)

Reply:
> only LCP before navst

that would never happen?

Reply:
> > only LCP before navst
> 
> that would never happen?

never say never when it comes to traces :)

Reply:
> > > only LCP before navst
> > 
> > 
> > that would never happen?
> 
> never say never when it comes to traces :)

excluding weird trace stuff, maybe one case that's possible is if the `passConfig.blankPage` is set to something with content on it...if it paints after that page loads and after we start tracing but before we navigate, it could end up in the trace.

Regardless, navstart is a reasonable cut off in the new trace-processor loop and we really would never want one that popped up somehow before navStart, so it's good to test.